### PR TITLE
Teach the analyzer about hidden columns

### PR DESCRIPTION
**Pull request by rjmac**

### CODE REVIEW

File: soql-analyzer/src/main/scala/com/socrata/soql/analyzer2/UnparsedTableMap.scala [hunk 1]
```diff
@@ -69,13 +74,20 @@ object UnparsedTableMap {
...
             desc match {
               case UnparsedTableDescription.Dataset(_name, _canonicalName, schema, ordering) =>
-                mocktablefinder.D(schema.valuesIterator.map { case NameEntry(n, t) => n.name -> t }.toSeq : _*)
-              case UnparsedTableDescription.Query(scope, canonicalName, basedOn, soql, parameters) =>
+                val base =
+                  mocktablefinder.D(schema.valuesIterator.map { case TableDescription.DatasetColumnInfo(n, t, _) => n.name -> t }.toSeq : _*).
+                    withHiddenColumns(schema.valuesIterator.filter(_.hidden).map(_.name.name).toSeq : _*)
+                ordering.foldLeft(base) { (base, ordering) =>
+                  base.withOrdering(schema(ordering.column).name.name, ordering.ascending)
+                }
```
Comment (rjmac) [inline soql-analyzer/src/main/scala/com/socrata/soql/analyzer2/UnparsedTableMap.scala:82]: This was just left out before.  Oops!

**Comment by rjmac** [comment]
Oh right, scala 2.12/13 difference.

File: soql-analyzer/src/main/scala/com/socrata/soql/analyzer2/SoQLAnalyzer.scala [hunk 2]
```diff
@@ -581,11 +647,12 @@ class SoQLAnalyzer[RNS, CT, CV](typeInfo: TypeInfo2[CT, CV], functionInfo: Funct
 
                 val useQuery =
                   new Context(udfScope, Some(udfCanonicalName), Environment.empty, innerUdfParams)
-                    .analyzeStatement(None, parsed, None)
+                    .analyzeStatement(None, parsed, None, hiddenColumns)
 
                 FromStatement(
                   Select(
                     Distinctiveness.Indistinct,
+                    // note: we're adding the column mask here
```
Comment (rjmac) [inline soql-analyzer/src/main/scala/com/socrata/soql/analyzer2/SoQLAnalyzer.scala:655]: This is actually no longer true, going to remove the comment.  (We're adding the column mask - what I called hiddenColumns before realizing that was a terrible name and changing it - a few lines upward)

**Review by saraid — APPROVED** [review]
I think I understand what's going on.

**Pull request merged**